tests/scripts/thread-cert/border_router/MATN_15_ChangeOfPrimaryBBRTriggersRegistration.py

```diff
@@ -0,0 +1,203 @@
+#
+
+import logging
+import unittest
+
+import pktverify
+from pktverify import packet_verifier, packet_filter, consts
+from pktverify.consts import MA1, MA2
+import config
+import thread_cert
+
+# Test description:
+# The purpose of this test case is to verify that a Primary BBR implements MLDv2
+# to receive IPv6 multicasts from upstream sources. Execute this test case with
+# the DUT in each of the following roles:
+#
+# Topology:
+#
+#
+#    ----------------(eth)--------------------
+#           |
+#         BR_1 (Leader) -----BR_2
+#           |                  |
+#           |                  |
+#           |                  |
+#         Router_1-------------+
+#           |
+#           |
+#          TD
+#
+
+BR_1 = 1
+BR_2 = 2
+ROUTER_1 = 3
+TD = 4
+
+REG_DELAY = 10
+
+
+class MATN_15_ChangeOfPrimaryBBRTriggersRegistration(thread_cert.TestCase):
+    USE_MESSAGE_FACTORY = False
+
+    TOPOLOGY = {
+        BR_1: {
+            'name': 'BR_1',
+            'is_otbr': True,
+            'allowlist': [BR_2, ROUTER_1],
+            'version': '1.2',
+            'router_selection_jitter': 2,
+        },
+        BR_2: {
+            'name': 'BR_2',
+            'is_otbr': True,
+            'allowlist': [BR_1, ROUTER_1],
+            'version': '1.2',
+            'router_selection_jitter': 2,
+        },
+        ROUTER_1: {
+            'name': 'Router_1',
+            'allowlist': [BR_1, BR_2, TD],
+            'version': '1.2',
+            'router_selection_jitter': 2,
+            'partition_id': 1,
+        },
+        TD: {
+            'name': 'TD',
+            'allowlist': [ROUTER_1],
+            'version': '1.2',
+            'router_selection_jitter': 2,
+            'partition_id': 1,
+        },
+    }
+
+    def test(self):
+        br1 = self.nodes[BR_1]
+        br2 = self.nodes[BR_2]
+        router1 = self.nodes[ROUTER_1]
+        td = self.nodes[TD]
+
+        br1.set_backbone_router(reg_delay=REG_DELAY, mlr_timeout=consts.MLR_TIMEOUT_MIN)
+        br1.start()
+        self.simulator.go(5)
+        self.assertEqual('leader', br1.get_state())
+        self.assertTrue(br1.is_primary_backbone_router)
+
+        br2.set_backbone_router(reg_delay=REG_DELAY, mlr_timeout=consts.MLR_TIMEOUT_MIN)
+        router1.start()
+        self.simulator.go(5)
+        self.assertEqual('router', router1.get_state())
+
+        td.start()
+        self.simulator.go(5)
+        self.assertEqual('router', td.get_state())
+
+        br2.start()
+        self.simulator.go(5)
+        self.assertEqual('router', br2.get_state())
+        self.assertFalse(br2.is_primary_backbone_router)
+
+        # TD registers for a multicast address, MA1.
+        td.add_ipmaddr(MA1)
+        self.simulator.go(20)
+
+        # 1. Disable BR_1 such that BR_2 becomes the Primary BBR for the Thread
+        # Network.
+        br1.disable_backbone_router()
+        br1.thread_stop()
+        br1.interface_down()
+        self.simulator.go(300)
```

Comment:
My understanding is, in this duration we need to wait for BR_2 to become PBBR, instead of waiting for router_1 to register multicast again. Thus it has nothing to do with MLR_TIMEOUT_MIN.

Reply:
Thanks for the tip. I misunderstood this step. 

Should we use a constant (e.g. NETWORK_ID_TIMEOUT = 120) rather than hard coding?